File: js/fizz-buzz.js
/*
  This variation of FizzBuzz users counters to keep track, meaning no division!
  The downside will be that you cannot turn this in to a function 
  that returns fizzbuzz for a given number.
  The upside is no modulus operations will make it incredibly fast.
*/

const fizzBuzz = (to) => {
  let fizz = 3;
  let buzz = 5;
  let counter = 0;

  while (counter < to) {
    counter++;

    console.log(
      `${!--fizz ? "fizz" : ""}${!--buzz ? "buzz" : ""}${
        fizz && buzz ? counter : ""
      }`
    );
    if (!fizz) fizz = 3;
    if (!buzz) buzz = 5;
  }
};

fizzBuzz(Number(process.argv[2] || 20));

File: python/fizz-buzz.py
# This is the classic build a string method in Python using the modulus operation

to = int(input("Range to: "))
nums = range(1, to + 1)

def fizzBuzz(num):
  string = ''
  if num % 3 == 0:
    string += 'fizz'
  if num % 5 == 0:
    string += 'buzz'
  if len(string) == 0:
    string = num

  return string

for num in nums:
  print(fizzBuzz(num))

